mysql-replication/azuredeploy.json

```diff
@@ -0,0 +1,508 @@
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "dnsName": {
+            "type": "string",
+            "metadata": {
+                "description": "Connect to your cluster using dnsName.location.cloudapp.azure.com"
+            }
+        },
+        "vmUserName": {
+            "type": "string",
+            "metadata": {
+                "description": "user name to ssh to the VMs"
+            }
+        },
+        "vmPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "password to ssh to the VMs"
+            }
+        },
+        "mysqlRootPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "mysql root user password"
+            }
+        },
+        "mysqlReplicationPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "mysql replication user password"
+            }
+        },
+        "mysqlProbePassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "mysql probe password"
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A1",
+            "metadata": {
+                "description": "size for the VMs"
+            }
+        },
+        "storageAccountType": {
+            "type": "string",
+            "defaultValue": "Standard_LRS",
+            "metadata": {
+                "description": "Storage account type for the cluster"
+            }
+        },
+        "virtualNetworkName": {
+            "type": "string",
+            "defaultValue": "mysqlvnet",
+            "metadata": {
+                "description": "Virtual network name for the cluster"
+            }
+        },
+        "dbSubnetName": {
+            "type": "string",
+            "defaultValue": "dbsubnet",
+            "metadata": {
+                "description": "subnet name for the MySQL nodes"
+            }
+        },
+        "vnetAddressPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "IP address in CIDR for virtual network"
+            }
+        },
+        "dbSubnetAddressPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.1.0/24",
+            "metadata": {
+                "description": "IP address in CIDR for db subnet"
+            }
+        },
+        "vmIPPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.1.",
+            "metadata": {
+                "description": "IP address prefix for VMs must be available in db subnet"
+            }
+        },
+        "vmIPStartAddress": {
+            "type": "int",
+            "defaultValue": 4,
+            "metadata": {
+                "description": "start IP address for VMs"
+            }
+        },
```

Comment:
All parameters below seem like they are needed for this scenario and could cause issues if other values are entered. Can these become variables to avoid issues and have the user enter only the most necessary parameters?